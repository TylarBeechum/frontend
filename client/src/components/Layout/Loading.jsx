import React from 'react';
import Loader from 'react-loader-spinner';
import styled from 'styled-components';

const Loading = () => {
  return (
    <LoadingContainer>
      <Loader type="RevolvingDot" color="#7D4CDB" height={100} width={100} />
    </LoadingContainer>
  );
};

const LoadingContainer = styled.div`
  display: flex;
  min-height: 100vh;
  background: #f8f8f8;
  align-items: center;
  justify-content: center;
`;

export default Loading;
