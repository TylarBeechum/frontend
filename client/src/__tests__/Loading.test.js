/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import Loading from '../components/Layout/Loading';

describe('<Loading />', () => {
  describe('should render the Loading component successfully without crashing', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(<Loading />);
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(<Loading />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('should render the Loader component to the screen successfully', () => {
    it('tests that the loader component is rendered successfully', () => {
      const { getByTestId } = renderWithRouterAndProviders(<Loading />);
      const loadingAnimation = getByTestId('loader');
      expect(loadingAnimation).toBeDefined();
    });
  });
});
