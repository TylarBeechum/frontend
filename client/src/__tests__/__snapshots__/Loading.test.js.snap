// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <div
    className="sc-EHOje egqVQK"
    data-testid="loader"
  >
    <div
      aria-busy="true"
      className=""
      style={Object {}}
    >
      <svg
        aria-label="audio-loading"
        height={100}
        version="1.1"
        width={100}
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <circle
          cx="50"
          cy="50"
          fill="none"
          r="44"
          stroke="#7D4CDB"
          strokeWidth="4"
          style={
            Object {
              "opacity": 0.5,
            }
          }
        />
        <circle
          cx="8"
          cy="54"
          fill="#7D4CDB"
          r="6"
          stroke="#7D4CDB"
          strokeWidth="3"
        >
          <animateTransform
            attributeName="transform"
            dur="2s"
            from="0 50 48"
            repeatCount="indefinite"
            to="360 50 52"
            type="rotate"
          />
        </circle>
      </svg>
    </div>
  </div>
</div>
`;
